src/components/projects/OrderedListToolbarItem.tsx

```diff
@@ -0,0 +1,251 @@
+  const isEnabled = !config.isEnabled || config.isEnabled(state)
+
+  /**
+   *  When choosing a list type will do one of these options:
+   *  * if text is selected will wrap it with an order list
+   *  * if the selection is on list will change just the list type and
+   *    **in case it's nested list will change just the lists at the same level**
+   */
+  const handleClick = useCallback(
+    (style) => {
+      if (!dispatch) {
+        return
+      }
+
+      const type = style.type
+      const { $from, $to } = state.selection
+      const range = $from.blockRange($to)
+      const isListNode =
+        range &&
+        schema.nodes.ordered_list.compatibleContent(
+          $from.node(range.depth - 1).type
+        )
+
+      if (!isListNode) {
+        const options = config.options
+        if (options) {
+          options[type](state, dispatch, { listStyleType: type })
+          view && view.focus()
+        }
+      } else {
+        updateListStyle(state, dispatch, range, type)
+      }
+    },
+    [config, dispatch, state, view]
+  )
+
+  return (
+    <ToolbarItem>
+      <ListButton
+        title={config.title}
+        data-active={config.isActive && config.isActive(state)}
+        disabled={!isEnabled}
+        onMouseDown={(event) => {
+          event.preventDefault()
+          config.run(state, dispatch)
+          view && view.focus()
+        }}
+      >
+        <OrderedList />
+      </ListButton>
+      <ListStyleSelector
+        disabled={!isEnabled}
+        handleClick={handleClick}
+        styles={[
+          { items: ['1.', '2.', '3.'], type: 'order' },
+          { items: ['A.', 'B.', 'C.'], type: 'alpha-upper' },
+          { items: ['a.', 'b.', 'c.'], type: 'alpha-lower' },
+          { items: ['I.', 'II.', 'III.'], type: 'roman-upper' },
+          { items: ['i.', 'ii.', 'iii.'], type: 'roman-lower' },
+        ]}
+      />
+    </ToolbarItem>
+  )
+}
+
+/**
+ *  Will use node depth as an indicator for the list level, and `sharedDepth` will
+ *  give us all depths for the selection so we can find the parent list
+ *  After that will look at the nodes in the parent list and just update
+ *  the list that had the same level to the target level user has selected
+ */
+export const updateListStyle = (
+  state: EditorState,
+  dispatch: (tr: Transaction) => void,
+  range: NodeRange,
+  type: string
+) => {
+  const { $from } = state.selection
+  const tr = state.tr
+  const targetLevel = range.depth - 1
+  const selectionDepth = $from.sharedDepth(state.selection.to)
+  const parentListDepth =
+    [...Array(selectionDepth)].findIndex((_, index) => {
+      const node = $from.node(index + 1)
+      return (
+        node.type === schema.nodes.ordered_list ||
+        node.type === schema.nodes.bullet_list
+      )
+    }) + 1
+  const parentListNode = $from.node(parentListDepth)
+  const parentListPos = $from.before(parentListDepth)
+
+  if (parentListDepth === targetLevel) {
+    replaceToOrderList(
```

Comment:
replaceToOrderList is one argument away from being a pure function declared out this function scope. I know it's copied so this is optional for this PR.